src/main/kotlin/dev/proxyfox/pluralkt/dispatch/EventDispatcher.kt

```diff
@@ -0,0 +1,25 @@
+package dev.proxyfox.pluralkt.dispatch
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+import kotlin.coroutines.CoroutineContext
+
+class EventDispatcher<E> : CoroutineScope {
+    override val coroutineContext: CoroutineContext = SupervisorJob() + Dispatchers.Default
+
+    val events = MutableSharedFlow<E>()
+
+    suspend inline fun <reified T : E> on(scope: CoroutineScope = this, noinline consumer: suspend T.() -> Unit): Job {
+        return events.buffer().filterIsInstance<T>().onEach { event ->
+            scope.launch {
+                runCatching {
+                    consumer(event)
+                }.onFailure {
+                    it.printStackTrace()
```

Comment:
Maybe a more advanced error handler would be more ideal.

At least log to SLF4J.

Reply:
Fixed in latest commit